lint/unusedParam_checker.go

```diff
@@ -0,0 +1,60 @@
+package lint
+
+import (
+	"go/ast"
+)
+
+func init() {
+	addChecker(unusedParamChecker{}, &ruleInfo{})
+}
+
+type unusedParamChecker struct {
+	baseFuncDeclChecker
+}
+
+func (c unusedParamChecker) New(ctx *context) func(*ast.File) {
+	return wrapFuncDeclChecker(&unusedParamChecker{
+		baseFuncDeclChecker: baseFuncDeclChecker{ctx: ctx},
+	})
+}
+
+func (c *unusedParamChecker) CheckFuncDecl(decl *ast.FuncDecl) {
+	params := decl.Type.Params
+	if params == nil || params.NumFields() == 0 {
+		return
+	}
+
+	// collect all params to map
+	objToIdent := make(map[*ast.Object]*ast.Ident)
+	for _, p := range params.List {
+		if len(p.Names) == 0 {
+			c.warnUnnamed(p)
+			break
```

Comment:
I believe it can be more sensible to do a `return` here?

Reply:
Yep, can be. Good catch.